src/lib/logical_query_plan/join_node.cpp

```diff
@@ -123,6 +123,27 @@ std::shared_ptr<LQPUniqueConstraints> JoinNode::unique_constraints() const {
   return unique_constraints;
 }
 
+std::vector<FunctionalDependency> JoinNode::pass_functional_dependencies() const {
+  /**
+   * Due to the logic of joins, we might lose several unique constraints in this node. Consequently, upper nodes
+   * will have less unique constraints to derive functional dependencies from.
+   * Consequently, we create functional dependencies from both input nodes to manually pass them up the LQP tree.
```

Comment:
Can we create FDs in the case of 1:1 and 1:n joins? Not saying that we have to add it right away, but if yes, we probably should point it out.

Reply:
Do you think of the following:

Two tables: 
* A, B
* C, D

Join on "B = C" should result in
* B -> C
* C -> B

Reply:
Correct. Also, if C is unique, B->D should also hold. Which, in turn makes me realize that we have not talked about transitive FDs yet. Again - not something that we should do in this PR, but we should highlight this concrete case (B->C and possible B->D) for the Join and in `functional_dependency.hpp`. For the latter, we might suggest that a future method `fds_apply(fds, dependent, dependee)` could take a set of FDs and two expressions to see if `dependee` is dependent on `dependent`. Basically what we are doing by hand for now. That could include checking for transitive FDs.

Reply:
Very useful ideas. I added notes for both of them.